src/diagnostics_component/utils/file_upload_utils/src/blob_storage_helper.cpp

```diff
@@ -0,0 +1,105 @@
+/**
+ * @file blob_storage_helper.cpp
+ * @brief Implements the interface for interacting with Azure Blob Storage
+ *
+ * @copyright Copyright (c) 2021, Microsoft Corp.
+ */
+#include "blob_storage_helper.hpp"
+
+#include <azure_c_shared_utility/string_token.h>
+#include <azure_c_shared_utility/urlencode.h>
+#include <cstring>
+#include <fstream>
+
+/**
+ * @brief Creates the blob storage client using the information in @p blobInfo and then constructs the object
+ * @param blobInfo information related to the blob storage account
+ * @param maxConcurrency the maximum number of threads that can be working at one time for the client connection
+ */
+AzureBlobStorageHelper::AzureBlobStorageHelper(const BlobStorageInfo& blobInfo, const unsigned int maxConcurrency)
+{
+    if (blobInfo.storageSasCredential == nullptr)
+    {
+        throw std::invalid_argument("Container name invalid");
+    }
+
+    // NOLINTNEXTLINE (modernize-make-unique): There is no std::make_unique in c++11
+    client = std::make_unique<Azure::Storage::Blobs::BlobContainerClient>(STRING_c_str(blobInfo.storageSasCredential));
+}
+
+/**
+ * @brief strips the file name from the @p filePath
+ * @param fileName a properly formed name for the file who's path is being constructed- must include extension if the file has one
+ * @param directoryPath a properly formed path to the directory where the file exists
+ * @returns the filePath, or ""
+ */
+std::string
+AzureBlobStorageHelper::CreatePathFromFileAndDirectory(const std::string& fileName, const std::string& directoryPath)
+{
+    if (fileName.empty() || directoryPath.empty())
+    {
+        throw std::runtime_error("Invalid Arguments to CreatePathFromFileAndDirectory");
```

Comment:
std::invalid_argument like on line 23 seems more appropriate. Also can shorten the string and just use `__FUNCTION__`, e.g.
```cpp
throw std::invalid_argument(__FUNCTION__)```